crates/bevy_render/src/render_task/bind.rs

```diff
@@ -0,0 +1,154 @@
+use crate::render_resource::{
+    BindGroupLayoutEntryBuilder, Buffer, IntoBindGroupLayoutEntryBuilder, IntoBinding, TextureView,
+};
+use bevy_derive::Deref;
+use wgpu::{
+    BindingResource, BindingType, BufferBindingType, StorageTextureAccess, TextureViewDimension,
+};
+
+/// Corresponds to `var<storage, read> my_buffer: T` in a WGSL shader.
+#[derive(Clone, Deref)]
+pub struct StorageBufferReadOnly<'a>(pub &'a Buffer);
+
+/// Corresponds to `var<storage, read_write> my_buffer: T` in a WGSL shader.
+#[derive(Clone, Deref)]
+pub struct StorageBufferReadWrite<'a>(pub &'a Buffer);
+
+/// Corresponds to `var my_texture: texture_2d<T>` in a WGSL shader.
+#[derive(Clone, Deref)]
+pub struct SampledTexture<'a>(pub &'a TextureView);
+
+/// Corresponds to `var my_texture: texture_storage_2d<F, read_write>` in a WGSL shader.
+#[derive(Clone, Deref)]
+pub struct StorageTextureReadWrite<'a>(pub &'a TextureView);
+
+/// Corresponds to `var my_texture: texture_storage_2d<F, write>` in a WGSL shader.
+#[derive(Clone, Deref)]
+pub struct StorageTextureWriteOnly<'a>(pub &'a TextureView);
+
+/// Corresponds to `var my_texture: texture_storage_2d<F, read>` in a WGSL shader.
+#[derive(Clone, Deref)]
+pub struct StorageTextureReadOnly<'a>(pub &'a TextureView);
+
+/// Corresponds to `var my_texture: texture_storage_2d<F, atomic>` in a WGSL shader.
+#[derive(Clone, Deref)]
+pub struct StorageTextureAtomic<'a>(pub &'a TextureView);
```

Comment:
I'm not keen on this. Adding separate types for all combinations of buffer and texture binding types feel 'undesirable'?

Reply:
Why so? You need to specify the bindings types _somewhere_.